### PR TITLE
Allow anonymous petition signing 

### PR DESCRIPTION
**Pull request by esirK**
## Description

- Allows anonymous petition signing 
- Adds versioning to API endpoints.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

#### Sign a petition
Screenshot 2021-07-09 at 11 22 20

#### View Signed pettition
Screenshot 2021-07-09 at 11 10 27

#### Show anonymous signing 
Screenshot 2021-07-09 at 17 15 51

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by esirK** [comment]
Hello @kilemensi , @KhadijaMahanga  any better naming suggestions for the `petitions` and `signatures` endpoints? Right now I'm exposing them directly after the `v1/` I would have preferred something close to the `v1/accounts/users/` and `v1/accounts/organizations/` format

File: actnow/petitions/models.py [hunk 1]
```diff
@@ -63,6 +63,9 @@ class Signature(TimestampedModelMixin):
         max_length=255,
         blank=True,
     )
+    sign_anonymously = models.BooleanField(
```
Comment (kilemensi) [inline actnow/petitions/models.py]:
```suggestion
    anonymous = models.BooleanField(
```

Signature means it has already been signed so it's more of was it anonymously signed?

File: actnow/urls.py [hunk 2]
```diff
@@ -18,9 +18,9 @@
 
 urlpatterns = [
     path("admin/", admin.site.urls),
-    path("accounts/", include("actnow.accounts.urls")),
-    path("petitions/", include("actnow.petitions.urls")),
-    path("profiles/", include("actnow.profiles.urls")),
+    path("v1/accounts/", include("actnow.accounts.urls")),
+    path("v1/", include("actnow.petitions.urls")),
+    path("v1/profiles/", include("actnow.profiles.urls")),
```
Comment (kilemensi) [inline actnow/urls.py]: Lets do this on a different PR.

**Review by kilemensi — COMMENTED** [review]
👍🏾

File: actnow/petitions/serializers.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 
 class SignatureSerializer(serializers.ModelSerializer):
-    signatory = UserSerializer(read_only=True)
+    signatory = serializers.HiddenField(default=serializers.CurrentUserDefault())
```
Comment (kilemensi) [inline actnow/petitions/serializers.py]: What does this mean?

Comment (esirK) [reply actnow/petitions/serializers.py]: the signature will automatically belong to the logged-in user.

Comment (esirK) [reply actnow/petitions/serializers.py]: removes the need to send a user id while signing a petition.

Comment (kilemensi) [reply actnow/petitions/serializers.py]: Yes but [docs](https://www.django-rest-framework.org/api-guide/validators/#advanced-field-defaults) says using `HiddenField` means the value will not be there in the output representation. This is what I'm asking what does it actually mean i.e. does it mean from signature output we won't be able to tell who signed it?

Comment (esirK) [reply actnow/petitions/serializers.py]: Yes, with this, user data won't be returned. Right now, a user can only access their signatures so adding the user details initially looked like overkill.   
But. If we want the ability to display all signatures/petitions and the people who did the signing then I can look for an alternative way to achieve the above objective.

Comment (kilemensi) [reply actnow/petitions/serializers.py]: I think the most common use of signatures is when someone is viewing a petition. These users are usually not the ones who created the petition and most probably they don't even have an account at this point e.g. I can view this [petition](https://www.change.org/p/south-african-post-office-postofficesa-and-icasa-icasa-org-we-have-right-to-choose-who-delivers-our-small-parcels?source_location=petitions_browse) with signatures without login.

Now with this design, when we return a petition with signatures, will the info be visible or not?

Comment (esirK) [reply actnow/petitions/serializers.py]: >Now with this design, when we return a petition with signatures, will the info be visible or not?

It won't. I'll revert to include the info

Comment (esirK) [reply actnow/petitions/serializers.py]: Fixed.

File: actnow/petitions/models.py [hunk 4]
```diff
@@ -63,6 +63,7 @@ class Signature(TimestampedModelMixin):
         max_length=255,
         blank=True,
     )
+    anonymous = models.BooleanField(verbose_name=_("Anonymous"), default=False)
```
Comment (kilemensi) [inline actnow/petitions/models.py]:
```suggestion
    anonymous = models.BooleanField(verbose_name=_("anonymous"), default=False)
```
Or why are we capitalizing it?

**Review by kilemensi — APPROVED** [review]
👍🏾 

----

We need to handle when anonymous=True to hide user details in the Serializer.

**Review by KhadijaMahanga — APPROVED** [review]
@esirK this lokks good. 

Though I am not sure why we want an anonymous person to sign petitions. Will they add at least their email address?

**Comment by esirK** [comment]
> @esirK this lokks good.
> 
> Though I am not sure why we want an anonymous person to sign petitions. Will they add at least their email address?

A user must be authenticated in order to signs a petition. The only thing this allows is if they choose to hide their details from being shown on the petition page.

**Comment by KhadijaMahanga** [comment]
@esirK Ahaa..got it

**Comment by esirK** [comment]
@kilemensi  
>We need to handle when anonymous=True to hide user details in the Serializer.

has been fixed.
